torch_glow/src/GlowFuser.cpp

```diff
@@ -51,140 +49,274 @@ sortReverseTopological(at::ArrayRef<torch::jit::Value *> inputs,
   return result;
 }
 
-bool canMerge(torch::jit::Node *node, isSupportFunc fn) {
-  return node->kind() == torch::jit::prim::Constant || fn(node);
-}
+bool canMerge(torch::jit::Node *node, IsSupportFunc fn,
+              torch::jit::Node *consumer) {
+  if (node->kind() == torch::jit::prim::Param) {
+    return false;
+  }
 
-#define REQ(cond, ignore, log_info)                                            \
-  if (!(cond)) {                                                               \
-    if (!(ignore)) {                                                           \
-      DLOG(ERROR) << (log_info);                                               \
-    }                                                                          \
-    return c10::nullopt;                                                       \
+  // Check that the node is supported
+  if (!(fn(node) || node->kind() == torch::jit::prim::Constant ||
+        node->kind() == torch::jit::prim::GetAttr)) {
+    return false;
   }
 
-// Check if a node is a known not-lowered operator, etc, a tensor generator.
-static bool isLogIgnoredOp(const std::string &op_name) {
-  static const std::unordered_set<std::string> white_list{"prim::GetAttr",
-                                                          "prim::Param"};
-  return white_list.count(op_name) > 0;
-}
+  // If the node is a producer (has a consumer), check that all non-tensor
+  // outputs are only consumed by the consumer.
+  for (torch::jit::Value *output : node->outputs()) {
+    if (output->type()->isSubtypeOf(torch::jit::TensorType::get())) {
+      continue;
+    }
 
-c10::optional<torch::jit::Node *> tryMerge(torch::jit::Node *consumer,
-                                           torch::jit::Node *producer,
-                                           torch::jit::AliasDb &aliasDb,
-                                           isSupportFunc fn, at::Symbol kind) {
+    // Producers can have non-tensor outputs as long as they are only consumed
+    // by consumer. Consumers cannot have non-tensor outputs.
+    if (consumer) {
+      for (auto use : output->uses()) {
+        if (use.user != consumer) {
```

Comment:
Why is this inequality?

Reply:
This is checking if the node is used anywhere outside of the fusion group